app/src/main/java/mysh/dev/gemcap/network/GeminiClient.kt

```
@@ -425,10 +507,34 @@ class GeminiClient(
+    ) {
+        val buffer = ByteArray(DEFAULT_BUFFER_SIZE)
+        var totalBytes = 0L
+        BufferedOutputStream(FileOutputStream(outputFile)).use { bos ->
+            while (true) {
+                val read = inputStream.read(buffer)
+                if (read == -1) break
+                totalBytes += read
+                if (totalBytes > MAX_FILE_RESPONSE_BODY_BYTES) {
+                    throw Exception(
+                        "Response body exceeds max file size of $MAX_FILE_RESPONSE_BODY_BYTES bytes"
+                    )
```

detekt (1.23.8)

[warning] Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)